src/python_testing/TC_IDM_3_2.py

```diff
@@ -0,0 +1,182 @@
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+import chip.clusters as Clusters
+
+from basic_composition_support import BasicCompositionTests
+from mobly import asserts
+from spec_parsing_support import build_xml_clusters
+
+class TC_IDM_3_2(MatterBaseTest, BasicCompositionTests):
+
+    @async_test_body
+    async def test_TC_IDM_3_2(self):
+
+        # Step 1
+
+        # TH selects a writeable attribute that is present on at least one endpoint.
+        # If no such attribute exists, skip this step. TH reads the attribute value and saves as `original`. TH sends a WriteRequestMessage to the DUT to write to the attribute on one endpoint.
+        #  It should set a value that is different than `original`.
+
+        # Verify that the DUT sends a WriteResponseMessage with any status except UNSUPPORTED_WRITE or DATA_VERSION_MISMATCH. If the Status is SUCCESS, verify the updated
+        #  value by sending a ReadRequestMessage for all affected paths. If the status is SUCCESS, send a WriteRequestMessage to set the value back to `original`.
+        self.user_params["use_pase_only"] = False
+        super().setup_class()
+        await self.setup_class_helper()
+        self.xml_clusters, self.problems = build_xml_clusters()
+        writable_attributes = []
+        for cluster_id in self.xml_clusters:
+            xml_cluster = self.xml_clusters[cluster_id]
+            attributes = xml_cluster.attributes
+            for attribute_id in attributes:
+                xml_attribute = attributes[attribute_id]
+                write_access = xml_attribute.write_access
+                if write_access:
+                    if cluster_id in Clusters.ClusterObjects.ALL_ATTRIBUTES:

```

Comment:
```suggestion
                    if cluster_id in Clusters.ClusterObjects.ALL_ATTRIBUTES and attribute_id in Clusters.ClusterObjects.ALL_ATTRIBUTES[cluster_id]:

```
^ that should let you eliminate the try loop below.